plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3Service.java

```diff
@@ -204,7 +204,11 @@ AmazonS3WithCredentials buildClient(final S3ClientSettings clientSettings) {
         builder.httpClientBuilder(buildHttpClient(clientSettings));
         builder.overrideConfiguration(buildOverrideConfiguration(clientSettings));
 
-        String endpoint = Strings.hasLength(clientSettings.endpoint) ? clientSettings.endpoint : DEFAULT_S3_ENDPOINT;
+        String s3Endpoint = Strings.hasText(clientSettings.region)
+            ? "s3." + Region.of(clientSettings.region).toString() + ".amazonaws.com"
+            : GLOBAL_S3_ENDPOINT;
```

Comment:
can we retrieve the endpoint from sdk itself , else this will break for CN and other non-classic partitions .